api/agent.go

```diff
@@ -730,11 +730,13 @@ func (a *Agent) ForceLeave(node string) error {
 	return nil
 }
 
-//ForceLeavePrune is used to have an a failed agent removed
-//from the list of members
-func (a *Agent) ForceLeavePrune(node string) error {
+//ForceLeaveWithOpts is used to have an a failed agent removed
+//from the list of members with specified additional arguments
+func (a *Agent) ForceLeaveWithOpts(node string, q *QueryOptions) error {
```

Comment:
After some internal discussion we decided that we shouldn't piggy-backoff of QueryOptions here.  QueryOptions should mostly be reserved for params that apply to multiple functions, rather than just for one (prune only applies to ForceLeave). 

Rather than passing in QueryOptions, the signature could instead be:
```go
func (a *Agent) ForceLeaveWithOpts(node string, datacenter string, prune bool) error
```

Then the params could be set like this:
```go
r.params.Set("prune", "")
r.params.Set("dc", datacenter)
```